From b05ab7f1b3d48e6165573cd515b3aae408837311 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:02 +0800
Subject: [PATCH 028/150] dt-bindings: mmc: phytium: Add bindings for Phytium
 MCI

This patch documents the DT bindings for Phytium MCI controller.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Iba5fc25d2604a910c28f7f0baa07d9a83b33b1e4
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/mmc/phytium,mci.yaml  | 67 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 68 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mmc/phytium,mci.yaml

diff --git a/Documentation/devicetree/bindings/mmc/phytium,mci.yaml b/Documentation/devicetree/bindings/mmc/phytium,mci.yaml
new file mode 100644
index 000000000000..a0748c70af37
--- /dev/null
+++ b/Documentation/devicetree/bindings/mmc/phytium,mci.yaml
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mmc/phytium,mci.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium Multimedia Card Interface controller
+
+description: |
+  The highspeed MMC host controller on Phytium SoCs provides an interface
+  for MMC, SD and SDIO types of memory cards.
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+allOf:
+  - $ref: "mmc-controller.yaml"
+
+properties:
+  compatible:
+    const: phytium,mci
+
+  reg:
+    maxItems: 1
+    description: mmc controller base registers.
+
+  interrupts:
+    maxItems: 1
+    description: mmc controller interrupt.
+
+  clocks:
+    maxItems: 1
+    description: phandles to input clocks.
+
+  clock-names:
+    items:
+      - const: phytium_mci_clk
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - clock-names
+
+examples:
+  - |
+    mmc0: mmc@28000000 {
+      compatible = "phytium,mci";
+      reg = <0x0 0x28000000 0x0 0x1000>;
+      interrupts = <GIC_SPI 72 IRQ_TYPE_LEVEL_HIGH>;
+      clocks = <&sysclk_1200mhz>;
+      clock-names = "phytium_mci_clk";
+      status = "disabled";
+    };
+
+    &mmc0 {
+      bus-width = <4>;
+      max-frequency = <50000000>;
+      cap-sdio-irq;
+      cap-sd-highspeed;
+      sd-uhs-sdr12;
+      sd-uhs-sdr25;
+      sd-uhs-sdr50;
+      no-mmc;
+      status = "ok";
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index b432cc3249de..b46b280222bb 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2532,6 +2532,7 @@ S:	Maintained
 W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
+F:	Documentation/devicetree/bindings/mmc/phytium,mci.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
 F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
-- 
2.47.0

